src/analysis/normed_space/weak_dual_of_normed_space.lean

```diff
@@ -0,0 +1,146 @@
+import analysis.normed_space.operator_norm
+
+/-!
+# Weak dual of normed space
+
+This file proves properties of the weak-* topology on the dual of a normed space.
+
+It is shown that the canconical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous, and as a
+consequence the weak-* topology is coarser than the topology induced by the dual norm (operator
+norm).
+
+## Main definitions
+
+The main definitions concern the canconical mapping `dual 𝕜 E → weak_dual 𝕜 E`.
+
+* `normed_space.dual.to_weak_dual` is a linear equivalence from `dual 𝕜 E`to `weak_dual 𝕜 E`.
+* `normed_space.dual.continuous_linear_map_to_weak_dual` is a continuous linear mapping from
+  `dual 𝕜 E` to `weak_dual 𝕜 E`.
+
+## Main results
+
+The main results concern the comparison of the operator norm topology on `dual 𝕜 E` and the weak-*
+topology on (it type synonym) `weak_dual 𝕜 E`.
+
+* `dual_norm_topology_le_weak_dual_topology` is the statement that the weak-* topology on the dual
+  of a normed space is coarser (not necessarily strictly) than the operator norm topology.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+Weak-* topology is defined generally in the file <topology.algebra.weak_dual_topology>.
+
+When `E` is a normed space, the duals `dual 𝕜 E` and `weak_dual 𝕜 E` are type synonyms with
+different topology instances.
+
+## References
+
+* https://en.wikipedia.org/wiki/Weak_topology#Weak-*_topology
+
+## Tags
+
+weak-star, weak dual
+
+-/
+
+noncomputable theory
+open filter
+open_locale topological_space
+
+section weak_star_topology_for_duals_of_normed_spaces
+/-!
+### Weak star topology on duals of normed spaces
+In this section, we prove properties about the weak-* topology on duals of normed spaces.
+We prove in particular that the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous,
+i.e., that the weak-* topology is coarser (not necessarily strictly) than the topology given
+by the dual-norm (i.e. the operator-norm).
+-/
+
+open normed_space
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E → weak_dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. -/
+def normed_space.dual.to_weak_dual : dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E :=
+linear_equiv.refl 𝕜 (E →L[𝕜] 𝕜)
+
+/-- For normed spaces `E`, there is a canonical map `weak_dual 𝕜 E → dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. Here it is implemented as the inverse of the linear
+    equivalence `normed_space.dual.to_weak_dual` in the other direction. -/
+def weak_dual.to_original_dual : weak_dual 𝕜 E ≃ₗ[𝕜] dual 𝕜 E :=
+normed_space.dual.to_weak_dual.symm
+
+@[simp] lemma weak_dual.coe_to_fun_eq_original_coe_to_fun (x' : dual 𝕜 E) :
+  (x'.to_weak_dual : E → 𝕜) = x' := rfl
+
+@[simp] lemma to_weak_dual_inj_iff (x' y' : dual 𝕜 E) :
+  x'.to_weak_dual = y'.to_weak_dual ↔ x' = y' := iff.rfl
```

Comment:
I don't really care about abusing definitional equality, but in this case the proof without the abuse is (I think) slightly more enlightening:
```suggestion
  x'.to_weak_dual = y'.to_weak_dual ↔ x' = y' :=
normed_space.dual.to_weak_dual.injective.eq_iff
```
Same for the next.  Also, can you please name it `to_weak_dual_eq_iff`, since that seems to be a standard naming pattern for this style of lemma?